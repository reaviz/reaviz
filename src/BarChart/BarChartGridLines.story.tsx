import React from 'react';
import { BarChart } from './BarChart';
import { categoryData } from 'reaviz-data-utils';
import { GridlineSeries, Gridline } from '@/common/Gridline';
import {
  Bar,
  BarLabel,
  BarSeries,
  GuideBar,
  HistogramBarSeries,
  MarimekkoBarSeries,
  RangeLines,
  StackedBarSeries,
  StackedNormalizedBarSeries
} from './BarSeries';

export default {
  tags: ['snapshot'],
  title: 'Charts/Bar Chart/Gridlines',
  component: BarChart,
  subcomponents: {
    BarSeries,
    StackedBarSeries,
    StackedNormalizedBarSeries,
    MarimekkoBarSeries,
    RangeLines,
    Bar,
    BarLabel,
    GuideBar,
    HistogramBarSeries
  }
};

export const AllAxes = () => (
  <BarChart
    width={350}
    height={250}
    data={categoryData}
    gridlines={<GridlineSeries line={<Gridline direction="all" />} />}
  />
);

export const XAxis = () => (
  <BarChart
    width={350}
    height={250}
    data={categoryData}
    gridlines={<GridlineSeries line={<Gridline direction="x" />} />}
  />
);

XAxis.story = {
  name: 'X-Axis'
};

export const YAxis = () => (
  <BarChart
    width={350}
    height={250}
    data={categoryData}
    gridlines={<GridlineSeries line={<Gridline direction="y" />} />}
  />
);

YAxis.story = {
  name: 'Y-Axis'
};
