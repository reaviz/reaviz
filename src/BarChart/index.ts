export * from './BarChart';
export * from './BarSeries';
export * from './HistogramBarChart';
export * from './MarimekkoChart';
export * from './StackedBarChart';
export * from './StackedNormalizedBarChart';
