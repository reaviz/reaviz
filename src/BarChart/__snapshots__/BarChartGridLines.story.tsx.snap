// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts/Bar Chart/Gridlines AllAxes smoke-test 1`] = `
<div class="_container_1u3dt_1"
     width="350"
     height="250"
     style="height: 250px; width: 350px;"
>
  <svg width="350"
       height="250"
       class="_svg_1u3dt_30 _barChart_sfjii_1"
       tabindex="0"
  >
    <g transform="translate(22, 0)">
      <g style="pointer-events: none;">
        <line y1="172"
              y2="172"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="129"
              y2="129"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="86"
              y2="86"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="43"
              y2="43"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="0"
              y2="0"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="8"
              x2="8"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="88"
              x2="88"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="168"
              x2="168"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="248"
              x2="248"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
      </g>
      <g transform="translate(0, 172)"
         visibility="visible"
      >
        <line x1="0"
              x2="328"
              y1="0"
              y2="0.00001"
              stroke-width="1"
              stroke="#8F979F"
        >
        </line>
        <g transform="translate(44, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              DLP
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              DLP
            </text>
          </g>
        </g>
        <g transform="translate(124, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              Malware
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              Malware
            </text>
          </g>
        </g>
        <g transform="translate(204, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              IDS
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              IDS
            </text>
          </g>
        </g>
        <g transform="translate(284, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              Phishing Attack
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              Phishing Attack
            </text>
          </g>
        </g>
      </g>
      <g transform="translate(0, 0)"
         visibility="visible"
      >
        <line x1="0"
              x2="0.00001"
              y1="172"
              y2="0"
              stroke-width="1"
              stroke="#8F979F"
        >
        </line>
        <g transform="translate(0, 172)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              0
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              0
            </text>
          </g>
        </g>
        <g transform="translate(0, 129)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              10
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              10
            </text>
          </g>
        </g>
        <g transform="translate(0, 86)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              20
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              20
            </text>
          </g>
        </g>
        <g transform="translate(0, 43)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              30
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              30
            </text>
          </g>
        </g>
        <g transform="translate(0, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              40
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              40
            </text>
          </g>
        </g>
      </g>
      <g>
        <rect height="172"
              width="328"
              opacity="0"
              cursor="auto"
        >
        </rect>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="DLP: 10"
                role="graphics-document"
                height="43px"
                width="72px"
                fill="url(#gradient-bar-series-ref-1-bar-undefined-0)"
                x="8"
                y="129"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-1-bar-undefined-0"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2d60e8"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2d60e8"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="Malware: 30"
                role="graphics-document"
                height="129px"
                width="72px"
                fill="url(#gradient-bar-series-ref-1-bar-undefined-1)"
                x="88"
                y="43"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-1-bar-undefined-1"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2c77e0"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2c77e0"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="IDS: 20"
                role="graphics-document"
                height="86px"
                width="72px"
                fill="url(#gradient-bar-series-ref-1-bar-undefined-2)"
                x="168"
                y="86"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-1-bar-undefined-2"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2b8dd8"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2b8dd8"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="Phishing Attack: 40"
                role="graphics-document"
                height="172px"
                width="72px"
                fill="url(#gradient-bar-series-ref-1-bar-undefined-3)"
                x="248"
                y="0"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-1-bar-undefined-3"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2aa1d1"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2aa1d1"
          >
          </stop>
        </linearGradient>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Charts/Bar Chart/Gridlines XAxis smoke-test 1`] = `
<div class="_container_1u3dt_1"
     width="350"
     height="250"
     style="height: 250px; width: 350px;"
>
  <svg width="350"
       height="250"
       class="_svg_1u3dt_30 _barChart_sfjii_1"
       tabindex="0"
  >
    <g transform="translate(22, 0)">
      <g style="pointer-events: none;">
        <line x1="8"
              x2="8"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="88"
              x2="88"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="168"
              x2="168"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line x1="248"
              x2="248"
              y1="0"
              y2="172"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
      </g>
      <g transform="translate(0, 172)"
         visibility="visible"
      >
        <line x1="0"
              x2="328"
              y1="0"
              y2="0.00001"
              stroke-width="1"
              stroke="#8F979F"
        >
        </line>
        <g transform="translate(44, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              DLP
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              DLP
            </text>
          </g>
        </g>
        <g transform="translate(124, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              Malware
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              Malware
            </text>
          </g>
        </g>
        <g transform="translate(204, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              IDS
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              IDS
            </text>
          </g>
        </g>
        <g transform="translate(284, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              Phishing Attack
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              Phishing Attack
            </text>
          </g>
        </g>
      </g>
      <g transform="translate(0, 0)"
         visibility="visible"
      >
        <line x1="0"
              x2="0.00001"
              y1="172"
              y2="0"
              stroke-width="1"
              stroke="#8F979F"
        >
        </line>
        <g transform="translate(0, 172)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              0
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              0
            </text>
          </g>
        </g>
        <g transform="translate(0, 129)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              10
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              10
            </text>
          </g>
        </g>
        <g transform="translate(0, 86)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              20
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              20
            </text>
          </g>
        </g>
        <g transform="translate(0, 43)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              30
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              30
            </text>
          </g>
        </g>
        <g transform="translate(0, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              40
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              40
            </text>
          </g>
        </g>
      </g>
      <g>
        <rect height="172"
              width="328"
              opacity="0"
              cursor="auto"
        >
        </rect>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="DLP: 10"
                role="graphics-document"
                height="43px"
                width="72px"
                fill="url(#gradient-bar-series-ref-11-bar-undefined-0)"
                x="8"
                y="129"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-11-bar-undefined-0"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2d60e8"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2d60e8"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="Malware: 30"
                role="graphics-document"
                height="129px"
                width="72px"
                fill="url(#gradient-bar-series-ref-11-bar-undefined-1)"
                x="88"
                y="43"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-11-bar-undefined-1"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2c77e0"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2c77e0"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="IDS: 20"
                role="graphics-document"
                height="86px"
                width="72px"
                fill="url(#gradient-bar-series-ref-11-bar-undefined-2)"
                x="168"
                y="86"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-11-bar-undefined-2"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2b8dd8"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2b8dd8"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="Phishing Attack: 40"
                role="graphics-document"
                height="172px"
                width="72px"
                fill="url(#gradient-bar-series-ref-11-bar-undefined-3)"
                x="248"
                y="0"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-11-bar-undefined-3"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2aa1d1"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2aa1d1"
          >
          </stop>
        </linearGradient>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Charts/Bar Chart/Gridlines YAxis smoke-test 1`] = `
<div class="_container_1u3dt_1"
     width="350"
     height="250"
     style="height: 250px; width: 350px;"
>
  <svg width="350"
       height="250"
       class="_svg_1u3dt_30 _barChart_sfjii_1"
       tabindex="0"
  >
    <g transform="translate(22, 0)">
      <g style="pointer-events: none;">
        <line y1="172"
              y2="172"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="129"
              y2="129"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="86"
              y2="86"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="43"
              y2="43"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
        <line y1="0"
              y2="0"
              x1="0"
              x2="328"
              class="_gridLine_5yx3q_1"
              stroke-dasharray="2 5"
              stroke-width="1"
              stroke="rgba(153, 153, 153, 0.5)"
              fill="none"
        >
        </line>
      </g>
      <g transform="translate(0, 172)"
         visibility="visible"
      >
        <line x1="0"
              x2="328"
              y1="0"
              y2="0.00001"
              stroke-width="1"
              stroke="#8F979F"
        >
        </line>
        <g transform="translate(44, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              DLP
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              DLP
            </text>
          </g>
        </g>
        <g transform="translate(124, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              Malware
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              Malware
            </text>
          </g>
        </g>
        <g transform="translate(204, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              IDS
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              IDS
            </text>
          </g>
        </g>
        <g transform="translate(284, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
          >
          </line>
          <g transform="translate(0, 10)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              Phishing Attack
            </title>
            <text transform="rotate(-60)"
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              Phishing Attack
            </text>
          </g>
        </g>
      </g>
      <g transform="translate(0, 0)"
         visibility="visible"
      >
        <line x1="0"
              x2="0.00001"
              y1="172"
              y2="0"
              stroke-width="1"
              stroke="#8F979F"
        >
        </line>
        <g transform="translate(0, 172)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              0
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              0
            </text>
          </g>
        </g>
        <g transform="translate(0, 129)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              10
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              10
            </text>
          </g>
        </g>
        <g transform="translate(0, 86)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              20
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              20
            </text>
          </g>
        </g>
        <g transform="translate(0, 43)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              30
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              30
            </text>
          </g>
        </g>
        <g transform="translate(0, 0)">
          <line stroke-width="1"
                stroke="#8F979F"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
          >
          </line>
          <g transform="translate(-10, 0)"
             font-size="11"
             font-family="sans-serif"
          >
            <title>
              40
            </title>
            <text transform
                  text-anchor="end"
                  alignment-baseline="middle"
                  fill="#8F979F"
            >
              40
            </text>
          </g>
        </g>
      </g>
      <g>
        <rect height="172"
              width="328"
              opacity="0"
              cursor="auto"
        >
        </rect>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="DLP: 10"
                role="graphics-document"
                height="43px"
                width="72px"
                fill="url(#gradient-bar-series-ref-21-bar-undefined-0)"
                x="8"
                y="129"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-21-bar-undefined-0"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2d60e8"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2d60e8"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="Malware: 30"
                role="graphics-document"
                height="129px"
                width="72px"
                fill="url(#gradient-bar-series-ref-21-bar-undefined-1)"
                x="88"
                y="43"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-21-bar-undefined-1"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2c77e0"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2c77e0"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="IDS: 20"
                role="graphics-document"
                height="86px"
                width="72px"
                fill="url(#gradient-bar-series-ref-21-bar-undefined-2)"
                x="168"
                y="86"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-21-bar-undefined-2"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2b8dd8"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2b8dd8"
          >
          </stop>
        </linearGradient>
        <g>
          <rect class
                mask
                rx="0"
                ry="0"
                tabindex="0"
                aria-label="Phishing Attack: 40"
                role="graphics-document"
                height="172px"
                width="72px"
                fill="url(#gradient-bar-series-ref-21-bar-undefined-3)"
                x="248"
                y="0"
                style="cursor: auto;"
          >
          </rect>
        </g>
        <linearGradient spreadmethod="pad"
                        id="gradient-bar-series-ref-21-bar-undefined-3"
                        x1="10%"
                        x2="10%"
                        y1="100%"
                        y2="0%"
        >
          <stop offset="0%"
                stop-opacity="0.3"
                stop-color="#2aa1d1"
          >
          </stop>
          <stop offset="80%"
                stop-opacity="1"
                stop-color="#2aa1d1"
          >
          </stop>
        </linearGradient>
      </g>
    </g>
  </svg>
</div>
`;
